json-fleece-openapi3/src/Fleece/OpenApi3.hs

```diff
@@ -398,99 +488,84 @@ mkInlineBodySchema raiseError schemaKey schemaMap schema =
-      Just OA.OpenApiObject ->
-        if IOHM.null (OA._schemaProperties schema)
-          then
-            mkAdditionalPropertiesMapSchema
-              raiseError
-              schemaKey
-              (\key itemSchema -> mkInlineBodySchema raiseError key schemaMap itemSchema)
-              (OA._schemaAdditionalProperties schema)
-          else do
-            (_moduleName, typeName) <- CGU.inferTypeForInputName CGU.Operation schemaKey
-            (fieldsSchemaMap, dataFormat) <-
-              mkOpenApiObjectFormat
-                CGU.Operation
-                schemaKey
-                typeName
-                schema
-
-            schemaTypeInfo <- CGU.inferSchemaInfoForTypeName typeName
-
-            let
-              codeGenType =
-                CGU.CodeGenType
-                  { CGU.codeGenTypeOriginalName = schemaKey
-                  , CGU.codeGenTypeName = typeName
-                  , CGU.codeGenTypeSchemaInfo = schemaTypeInfo
-                  , CGU.codeGenTypeDescription = NET.fromText =<< OA._schemaDescription schema
-                  , CGU.codeGenTypeDataFormat = dataFormat
-                  }
-
-              schemaEntry =
-                SchemaEntry
-                  { schemaOpenApiSchema = schema
-                  , schemaCodeGenType = codeGenType
-                  }
-
-              codeGenModules =
-                Map.insert
-                  (CGU.SchemaKey schemaKey)
-                  schemaEntry
-                  fieldsSchemaMap
-
-            pure $
-              SchemaTypeInfoWithDeps
-                { schemaTypeInfoDependent = schemaTypeInfo
-                , schemaTypeInfoDependencies = codeGenModules
-                }
-      Just s ->
-        raiseError $ "Inline " <> show s <> " schemas are not currently supported."
-      Nothing ->
-        raiseError "Inline schema doesn't have a type."
+    case OA._schemaItems schema of
+      Just (OA.OpenApiItemsObject (OA.Ref (OA.Reference itemRefKey))) -> do
+        itemSchemaInfo <- lookupCodeGenType itemRefKey
+        pure . schemaInfoWithoutDependencies . CGU.arrayTypeInfo $ itemSchemaInfo
+      Just (OA.OpenApiItemsObject (OA.Inline innerSchema)) ->
+        let
+          itemKey =
+            schemaKey <> "Item"
+        in
+          fmap
+            (fmapSchemaInfoAndDeps $ first CGU.arrayTypeInfo)
+            (mkInlineBodySchema raiseError itemKey schemaMap innerSchema)
+      otherItemType ->
+        raiseError $
+          "Unsupported schema array item type found: "
+            <> show otherItemType
+
+mkInlineArrayOneOfSchema ::
+  (forall a. String -> CGU.CodeGen a) ->
+  T.Text ->
+  SchemaMap ->
+  OA.Schema ->
+  CGU.CodeGen SchemaTypeInfoWithDeps
+mkInlineArrayOneOfSchema raiseError schemaKey schemaMap schema =
+  case OA._schemaItems schema of
+    Just (OA.OpenApiItemsObject (OA.Ref ref)) -> do
+      pure $
+        -- We don't currently have a good way to get the nullability of the referenced schema
```

Comment:
This is probably an issues we could fix because it leaves things in state that will generate code that doesn't match the OpenAPI description in a very subtle and non-obvious way.

Reply:
I added several tests and support for inline nullability, there are still some lingering edge cases (with other features we haven't fully covered yet like inline nested objects, or nested oneOf), but stuff like `nullable: true` with a `$ref` sibling doesn't even appear to be valid according to the spec, which I didn't realize at first.